test/mpi/configure.ac

```diff
@@ -624,6 +624,21 @@ AC_SUBST(mpix)
 # to noninst_PROGRAMS to skip building the tests when we do strict MPI test
 AM_CONDITIONAL([BUILD_MPIX_TESTS], [test "$enable_strictmpi" = "no"])
 
+# Many functions were deprecated in MPI-2 and deleted in MPI-3.
+# They are not present in implementations that have implemented MPI-3 deletions.
+#
+MPI_HAS_MPI3_DELETED=no
+#
+# If MPI is derived from MPICH and strictmpi not selected, let's test MPI-3 deleted features.
+if test "$MPI_IS_MPICH" = "yes" -a "$enable_strictmpi" = "no" ; then
+   MPI_HAS_MPI3_DELETED=yes
+fi
+AC_SUBST(MPI_HAS_MPI3_DELETED)
+
+# Use the conditional variable BUILD_MPI3_DELETED_TESTS to conditionally add these tests
+# to noninst_PROGRAMS to skip building the tests when we do strict MPI test
+AM_CONDITIONAL([BUILD_MPI3_DELETED_TESTS], [test "$enable_strictmpi" = "no"])
+
```

Comment:
How about combining it with `BUILD_MPIX_TESTS`, maybe renaming it to `BUILD_NONSTRICT_TESTS`?

Reply:
Yes. They have to use older testsuite releases or just disable strictmpi and xfail the tests they do not support, if they really need test those deleted APIs.

Reply:
Anyway, I guess we can merge this and clean it up later.